From cd7f23f1187144fca1ec653d3aff59af59f30e96 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 15 Feb 2024 12:42:33 +0200
Subject: [PATCH] [SW-167279] boards: its: remove security version and key idx from its

Currently, while generating fit some of the ITS files adds security
version and key idx to the fit. In the signing process, the script
checks where key idx is part of fit and if its available it will
update it with key idx user passed to sign the image. If key idx is
not found in fit, script goes and takes it from fit footer.

The problem of adding this information in fit occurs for computing
fit sha's which are used for TPM PCR extention. Ideally when
the signing script replaces the signature from key0 to key1 or key0
to key2 and viceversa, the hashes of the images (ppboot, preboot,
boot fit and linux fit) should not change and in CI, it uses the
same pcr binary. But since the key idx is added in fit, when it changes
from 0 to 1, it causes boot fit hash to change. So on system fw
reports new hash but the tpm tool compares it with pcr which has
hashes without key-idx value change.

To resolve above problem, this patch removes security version and key
idx values from fits where they are added as they are ideally taken
from fit footer.

Change-Id: I6eadda839867843ef87cef264c9f89f6e88af5c1
---

diff --git a/board/habanalabs/common/uboot-zephyr_reloc.its b/board/habanalabs/common/uboot-zephyr_reloc.its
index 5f0fed8..e823a4d 100644
--- a/board/habanalabs/common/uboot-zephyr_reloc.its
+++ b/board/habanalabs/common/uboot-zephyr_reloc.its
@@ -6,8 +6,6 @@
 	hl-zephyr-mgmt-version = "@HL_ZEPHYR_MGMT_VERSION@";
 	hl-zephyr-pid-version = "@HL_ZEPHYR_PID_VERSION@";
 	hl-device = "@DEV_PREFIX@";
-	hl-sec-key-idx = <@HL_KEY_INDEX@>;
-	hl-sec-version = <@HL_SEC_FW_VERSION@>;
 	description = "FIT image with U-Boot and Zephyr";
 	#address-cells = <2>;
 
diff --git a/board/habanalabs/common/uboot-zephyr_sram.its b/board/habanalabs/common/uboot-zephyr_sram.its
index 290f3a5..94ed2a7 100644
--- a/board/habanalabs/common/uboot-zephyr_sram.its
+++ b/board/habanalabs/common/uboot-zephyr_sram.its
@@ -6,8 +6,6 @@
 	hl-zephyr-mgmt-version = "@HL_ZEPHYR_MGMT_VERSION@";
 	hl-zephyr-pid-version = "@HL_ZEPHYR_PID_VERSION@";
 	hl-device = "@DEV_PREFIX@";
-	hl-sec-key-idx = <@HL_KEY_INDEX@>;
-	hl-sec-version = <@HL_SEC_FW_VERSION@>;
 	description = "FIT image with U-Boot and Zephyr";
 	#address-cells = <2>;
 
diff --git a/board/habanalabs/gaudi2/agent.its b/board/habanalabs/gaudi2/agent.its
index dd96ff3..f8523fa 100644
--- a/board/habanalabs/gaudi2/agent.its
+++ b/board/habanalabs/gaudi2/agent.its
@@ -10,8 +10,6 @@
 	hl-device = "@HL_DEV@";
 	description = "FIT image with agent, loaded from preboot";
 	#address-cells = <2>;
-	hl-sec-key-idx = <@HL_KEY_INDEX@>;
-	hl-sec-version = <@HL_SEC_FW_VERSION@>;
 
 	images {
 		agent {
